ui/app/utils/classes/poll-logger.js

```diff
@@ -0,0 +1,60 @@
+import Ember from 'ember';
+import queryString from 'npm:query-string';
+import { task, timeout } from 'ember-concurrency';
+
+const { Object: EmberObject, computed, assign } = Ember;
+
+export default EmberObject.extend({
+  url: '',
+  interval: 1000,
+  params: computed(() => ({})),
+  logFetch() {
+    Ember.assert(
+      'Loggers need a logFetch method, which should have an interface like window.fetch'
+    );
+  },
+
+  endOffset: null,
+
+  fullUrl: computed('url', 'endOffset', 'params', function() {
+    const endOffset = this.get('endOffset');
+    let additionalParams;
+    if (endOffset) {
+      additionalParams = {
+        origin: 'start',
+        offset: this.get('endOffset'),
+      };
+    } else {
+      additionalParams = {
+        origin: 'end',
+        offset: 50000,
```

Comment:
I wonder if you might wanna throw `50000` in a config file since it's used in several spots.